pkg/certificates/reconciler/controller_impl.go

```diff
@@ -0,0 +1,109 @@
+/*
+Copyright 2022 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package sample
+
+import (
+	context "context"
+	fmt "fmt"
+	reflect "reflect"
+	strings "strings"
+
+	zap "go.uber.org/zap"
+	corev1 "k8s.io/api/core/v1"
+	watch "k8s.io/apimachinery/pkg/watch"
+	informersv1 "k8s.io/client-go/informers/core/v1"
+	scheme "k8s.io/client-go/kubernetes/scheme"
+	v1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	record "k8s.io/client-go/tools/record"
+	client "knative.dev/pkg/client/injection/kube/client"
+	secretreconciler "knative.dev/pkg/client/injection/kube/reconciler/core/v1/secret"
```

Comment:
I think this still wires in the non-filtered informer indirectly via an import

https://github.com/knative/pkg/blob/8fa1a3f55bf5dc307c7515bb05cf03c9b39b311a/client/injection/kube/reconciler/core/v1/secret/controller.go#L36

Reply:
I'm only pulling in the `secretreconciler` to use the reconciler.go file which doesn't import the secret informer and allows me to pass in the the filtered lister defined in the above step.

Do you prefer if I have a new implementation of the reconciler here as well?